app/models/role_scoped_regions.rb

```diff
@@ -0,0 +1,73 @@
+module RoleScopedRegions
+  def facility_groups(role: :write)
+    roles = if role == :write
+      ["super_admin", "admin"]
+    else
+      ["super_admin", "admin", "analyst"]
+    end
+    sql = <<-SQL
+      SELECT facility_groups.* from facility_groups
+      INNER JOIN organizations
+        ON facility_groups.organization_id = organizations.id
+      INNER JOIN roles
+        ON roles.resource_type = 'Organization'
+        AND roles.resource_id = organizations.id
+      WHERE roles.user_id = :user_id
+        AND roles.name in (:roles)
+
+      UNION
+
+      SELECT facility_groups.* from facility_groups
+      INNER JOIN roles
+        ON roles.resource_type = 'FacilityGroup'
+        AND roles.resource_id = facility_groups.id
+      WHERE roles.user_id = :user_id
+        AND roles.name in (:roles)
+    SQL
+    ids = FacilityGroup.find_by_sql([sql, {roles: roles, user_id: id}]).pluck(:id)
+    FacilityGroup.where(id: ids)
```

Comment:
Here we re-find things by id so that we return an ActiveRecord Relation instead of an Array. 

There are gems and hacks to do a Union in plain ARel, but I'm not sure its really worth it...these queries should be very fast once we get the indexes in place.